src/CommunityCommons/javasource/communitycommons/actions/EncryptString.java

```diff
@@ -1,51 +0,0 @@
-// This file was generated by Mendix Studio Pro.
-//
-// WARNING: Only the following code will be retained when actions are regenerated:
-// - the import list
-// - the code between BEGIN USER CODE and END USER CODE
-// - the code between BEGIN EXTRA CODE and END EXTRA CODE
-// Other code you write will be lost the next time you deploy the project.
-// Special characters, e.g., é, ö, à, etc. are supported in comments.
-
-package communitycommons.actions;
-
-import communitycommons.StringUtils;
-import com.mendix.systemwideinterfaces.core.IContext;
-import com.mendix.webui.CustomJavaAction;
-
-/**
- * Decrypts an AES encrypted string.
- * The keylength should exactly be 16 characters (128 bit).
- */
-public class EncryptString extends CustomJavaAction<java.lang.String>

```

Comment:
I don't immediately see how this was deprecated. How are people informed about the deprecation and removal of this action (and the `DecryptString` action)?

Reply:
In the release notes :) I'm not aware of any way to deprecate a java action so that it shows up in studio pro

Reply:
Ah, that is unfortunate. In that case, I'll resolve this as we cannot do anything more right now.

Reply:
There is no "official" way to do it, but the Java actions in CC are "exposed in microflow action", so there is nothing stopping us from putting deprecated actions in a "Deprecated" category and documenting it as such.

Reply:
Good idea, we'll do that next time :)

Reply:
@reinouts Would you happen to know if there are plans to create an "official" way to do it? I think that if it was possible to make these deprecations show up in Studio Pro, it would benefit users because it would be a lot more transparent as to what is going on.

Though, I do not know how feasible or how much effort that will take.

Reply:
@mx-adrian I am not aware of such plans, but then again I'm not working on SP. Having said that: there is an existing way to show deprecations for various constructs in the model in the Warnings pane. If you work with the relevant Studio Pro team(s) my guesstimate is that it would be relatively low effort to add a 'deprecated' checkbox to Java actions and have them show up in the Deprecations list.